phoenix-core/src/main/java/org/apache/phoenix/coprocessor/ChildLinkMetaDataEndpoint.java

```diff
@@ -111,4 +143,197 @@ private PhoenixMetaDataCoprocessorHost getCoprocessorHost() {
+         */
+        private byte[] getChildParentLinkSysCatRowKey(byte[] childLinkRowKey) {
+            String[] childLinkRowKeyCols = new String(childLinkRowKey, StandardCharsets.UTF_8)
+                                                .split(NULL_DELIMITER);
+            checkArgument(childLinkRowKeyCols.length == 5);
+            String parentTenantId = childLinkRowKeyCols[0];
+            String parentSchema = childLinkRowKeyCols[1];
+            String parentTable = childLinkRowKeyCols[2];
+            String childTenantId = childLinkRowKeyCols[3];
+            String childFullName = childLinkRowKeyCols[4];
+
+            String parentFullName = SchemaUtil.getTableName(parentSchema, parentTable);
+            String childSchema = SchemaUtil.getSchemaNameFromFullName(childFullName);
+            String childTable = SchemaUtil.getTableNameFromFullName(childFullName);
+
+            String[] sysCatRowKeyCols = new String[] {childTenantId, childSchema, childTable,
+                                                        parentTenantId, parentFullName};
+            return String.join(NULL_DELIMITER, sysCatRowKeyCols).getBytes(StandardCharsets.UTF_8);
+        }
+
+
+        private void deleteIfAgedEnough(byte[] rowKey, long ts, Region region) throws IOException {
+            if ((EnvironmentEdgeManager.currentTimeMillis() - ts) > ageThreshold) {
+                Delete del = new Delete(rowKey);
+                Mutation[] mutations = new Mutation[]{del};
+                region.batchMutate(mutations);
+            }
+        }
+
+
+        private void markChildLinkVerified(byte[] rowKey, long ts, Region region) throws IOException {
+            Put put = new Put(rowKey, ts);
+            put.addColumn(emptyCF, emptyCQ, ts, VERIFIED_BYTES);
+            Mutation[] mutations = new Mutation[]{put};
+            region.batchMutate(mutations);
+        }
+
+        private void setSysCatViewHeaderRowScan() {
+            sysCatViewHeaderRowScan = new Scan();
+            sysCatViewHeaderRowScan.addColumn(TABLE_FAMILY_BYTES, TABLE_TYPE_BYTES);
+            SingleColumnValueFilter tableTypeFilter = new SingleColumnValueFilter(
+                    TABLE_FAMILY_BYTES, TABLE_TYPE_BYTES, CompareOperator.EQUAL,
+                    PTableType.VIEW.getSerializedValue().getBytes());
+            sysCatViewHeaderRowScan.setFilter(tableTypeFilter);
+        }
+
+    }
+
+    @Override
+    public Optional<RegionObserver> getRegionObserver() {
+        return Optional.of(this);
+    }
+
+    @Override
+    public RegionScanner postScannerOpen(ObserverContext<RegionCoprocessorEnvironment> c,
+                                         Scan scan, RegionScanner s) throws IOException {
+        if (scan.getAttribute(CHECK_VERIFY_COLUMN) == null) {
+            return s;
+        }
+        return new ChildLinkMetaDataScanner(c.getEnvironment(), scan, s);
```

Comment:
You need to wrap this around PagingRegionScanner to get paging to work. `return new ChildLinkMetaDataScanner(c.getEnvironment(), scan,
                new PagingRegionScanner(c.getEnvironment().getRegion(), s, scan));`